### PR TITLE
Fix #50 #51 #52 print button works. print modal new layout. success notif shows on print

### LINKED ISSUE #50: Create success notification that shows on print

### PR DESCRIPTION
**Pull request by shaunBoahCodes**
print button in search now shows print modal on click. Print modal has new layout with qr code and select all button. Created success notification that shows on print.

### CODE REVIEW

**Review by cweitat — CHANGES_REQUESTED** [review]
update your PR title; too long. 

content add in description

File: src/components/Modals/PrintModal.vue [hunk 1]
```diff
@@ -15,121 +15,105 @@ const updateLocalShowNotification = (value) => {
...
-onMounted(() => {
-  refs.value = Array(printOptions.length)
-    .fill(null)
-    .map(() => ref(null))
-})
-function setRef(index) {
-  return (el) => {
-    refs.value[index] = el
+watch(
+  () => props.showNotification,
```
Comment (cweitat) [inline src/components/Modals/PrintModal.vue:19]: dont't think this is needed in the first place. on dialog close, you could have just emit the event out to update the close status which is taken in by the props. makes no sense to store a prop into a variable and watch it when the props is already reactive

**Review by cweitat — CHANGES_REQUESTED** [review]

File: src/components/Modals/PrintModal.vue [hunk 2]
```diff
@@ -15,121 +15,105 @@ const updateLocalShowNotification = (value) => {
...
-      selectAll.value.checked = true
-    }, 0)
-  }
-)
 </script>
 
 <template>
-  <TransitionRoot v-if="notificationContent" as="template" :show="open">
+  <TransitionRoot as="template" :show="open">
     <Dialog as="div" class="relative z-10" @close="updateLocalShowNotification">
```
Comment (cweitat) [inline src/components/Modals/PrintModal.vue:117]: could have just emit false on close

**Review by cweitat — CHANGES_REQUESTED** [review]

File: src/components/Registration/Manual/ScanSearch.vue [hunk 3]
```diff
@@ -78,7 +73,18 @@ async function logErrors(promise) {
...
+    :show="showPrintedNotification"
+    @update-show="showPrintedNotification = false"
+  />
+  <div class="mx-auto grid grid-cols-1 xl:flex items-center gap-16 lg:w-3/4 h-full py-16">
+    <PrintModal
+      :key="componentKey"
+      :showNotification="showNotification"
+      :validQRCode="validQRCode"
+      @update:show-modal="showNotification = false"
+      @print="printFunction"
```
[the remark this answers is not in the document]
Comment (shaunBoahCodes) [reply src/components/Registration/Manual/ScanSearch.vue:86]: @cweitat which line or function are you referring to?

[the remark this answers is not in the document]
Comment (cweitat) [reply src/components/Registration/Manual/ScanSearch.vue:86]: line 53

File: src/components/Registration/Manual/ScanSearch.vue [hunk 4]
```diff
@@ -106,7 +111,14 @@ async function logErrors(promise) {
       <div class="w-full"></div>
     </div>
     <div class="grow">
-      <SearchAttendee />
+      <SearchAttendee
+        @print="
+          ($event) => {
+            showPrintModal = true
```
Comment (cweitat) [inline src/components/Registration/Manual/ScanSearch.vue:117]: why not just showprintmodal =true? console.log still needed?

File: src/components/Shared/SuccessNotification.vue [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+import { CheckCircleIcon } from '@heroicons/vue/24/outline'
+import { XMarkIcon } from '@heroicons/vue/20/solid'
+
+const props = defineProps({
+  showPrintedNotif: Boolean
+})
+const emit = defineEmits(['hidePrintedNotif'])
+
+watch(
+  () => props.showPrintedNotif,
```
Comment (cweitat) [inline src/components/Shared/SuccessNotification.vue:12]: please use full name

**Review by cweitat — CHANGES_REQUESTED** [review]

File: src/components/Registration/Kiosk/ScannerCamera.vue [hunk 6]
```diff
@@ -78,19 +73,23 @@ async function logErrors(promise) {
 </script>
 
 <template>
+  <SuccessNotification
+    :showPrintedNotif="showPrintedNotification"
+    @hide-printed-notif="showPrintedNotification = false"
```
Comment (cweitat) [inline src/components/Registration/Kiosk/ScannerCamera.vue:78]: stick with camel case

File: src/components/Registration/Kiosk/ScannerCamera.vue [hunk 7]
```diff
@@ -78,19 +73,23 @@ async function logErrors(promise) {
...
+  />
   <div class="h-full mx-auto max-w-7xl flex">
     <PrintModal
-      :showNotification="showNotification"
+      :key="componentKey"
+      :showPrintModal="showPrintModal"
       :validQRCode="validQRCode"
-      @update:show-modal="updateShowNotification"
+      @hide-modal="showPrintModal = false"
```
Comment (cweitat) [inline src/components/Registration/Kiosk/ScannerCamera.vue:85]: ccamel

**Review by cweitat — CHANGES_REQUESTED** [review]

File: src/components/Shared/SuccessNotification.vue (hunk 5 above)
Comment (cweitat) [reply src/components/Shared/SuccessNotification.vue:12]: Notification

File: src/components/Modals/PrintModal.vue [hunk 8]
```diff
@@ -1,136 +1,108 @@
...
-}
+const disableButton = ref(false)
+const selectedOptions = ref(['code', 'name', 'email', 'org', 'role'])
 
-const notificationContent = ref({
-  titleText: '',
-  messageText: '',
-  buttonText: ''
-})
+const titleText = ref(props.validQRCode ? 'Select items to print' : 'Error!')
```
Comment (cweitat) [inline src/components/Modals/PrintModal.vue:17]: this shouldn't be used like this. refer to computed property

File: src/components/Modals/PrintModal.vue [hunk 9]
```diff
@@ -1,136 +1,108 @@
...
+const messageText = ref(props.validQRCode ? '' : 'Please scan a valid QR code')
```
Comment (cweitat) [inline src/components/Modals/PrintModal.vue:18]: this shouldn't be used like this. refer to computed property

File: src/components/Modals/PrintModal.vue [hunk 10]
```diff
@@ -1,136 +1,108 @@
...
+      element.checked.value = false
+      selectedOptions.value = ['code']
     })
   } else {
-    printOptions.forEach((element, index) => {
-      refs.value[index].checked = false
-      selectedOptions.value.pop(element.id)
+    printOptions.slice(1).forEach((element) => {
+      element.checked.value = true
+      selectedOptions.value = ['code', 'name', 'email', 'org', 'role']
```
Comment (cweitat) [inline src/components/Modals/PrintModal.vue:96]: do note that this is not viable as the real list is always dynamic

**Review by cweitat — CHANGES_REQUESTED** [review]

File: src/components/Modals/PrintModal.vue (hunk 9 above)
Comment (shaunBoahCodes) [reply src/components/Modals/PrintModal.vue:18]: these cannot be computed because they have to be reassigned a new value in the print function

Comment (cweitat) [reply src/components/Modals/PrintModal.vue:18]: can be computed. computed trigger upon any value that is changed within the function
https://vuejs.org/guide/essentials/computed.html#basic-example

**Review by cweitat — APPROVED** [review]

**Comment by cweitat** [comment]
Unable to get Codacy to run due to overload. Merging for now

**Pull request merged**